cassiopeia/solver/DistanceSolver.py

```diff
@@ -75,6 +75,39 @@ def __init__(
         self.dissimilarity_function = dissimilarity_function
         self.add_root = add_root
 
+    def get_dissimilarity_map(
+        self, 
+        cassiopeia_tree: CassiopeiaTree,
+        layer: Optional[str] = None
+    ) -> pd.DataFrame: 
+        """Obtains or generates a matrix that is updated throughout the solve.
+
+        The highest-level method to obtain a dissimilarity map, which will be 
+        the matrix primarily used throughout the solve method. This matrix 
+        contains information for decisions related to clade formation, and 
+        will be updated at every iteration within the solve method. This method 
+        is not limited to outputting dissimilarity maps, but is instead 
+        deliberately designed to be overwritten to allow for similarity maps and
+        lambda matrices if needed.
+
+        Args:
+            cassiopeia_tree (CassiopeiaTree): Tree object from which the 
+                dissimilarity map is generated from
+            layer (Optional[str], optional): Layer storing the character matrix 
+                for solving. If None, the default character matrix is used in 
+                the CassiopeiaTree.
```

Comment:
Our codestyle guidelines don't necessitate putting type info into the Argument section like this. Is this a best practice we should consider? If not, I would rather not have type info like this in the Args section.

Reply:
i removed them for now